tools-local/tasks/BuildFPMToolPreReqs.Netstandard2_0.cs

```diff
@@ -0,0 +1,581 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// This task prepares the command line parameters for running a RPM build using FPM tool and also updates the copyright and changelog file tokens.
+    /// If parses various values from the config json by first reading it into a model and then builds the required string for parameters and passes it back.
+    /// 
+    /// </summary>
+    public class BuildFPMToolPreReqs : BuildTask
+    {
+        [Required]
+        public string InputDir { get; set; }
+        [Required]
+        public string OutputDir { get; set; }
+        [Required]
+        public string PackageVersion { get; set; }
+        [Required]
+        public string ConfigJsonFile { get; set; }
+        [Output]
+        public string FPMParameters { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                if (!File.Exists(ConfigJsonFile))
+                {
+                    throw new FileNotFoundException($"Expected file {ConfigJsonFile} was not found.");
+                }
+
+                // Open the Config Json and read the values into the model
+                TextReader projectFileReader = File.OpenText(ConfigJsonFile);
+                if (projectFileReader != null)
+                {
+                    string jsonFileText = projectFileReader.ReadToEnd();
+
+                    // TODO: Replace the use of the Utf8JsonReader with the Deserializer once it is available.
+                    ConfigJson configJson = GetConfigJson(jsonFileText);
```

Comment:
So we're opening the file, reading all of its contents into a string (which will involve converting from UTF8 to UTF16 if it's saved as UTF8), and then turning around and encoding that string into a UTF8 byte[]?  Isn't the point of having a UTF8-based JSON reader to avoid that?

Reply:
I was going to ask the same question, but then I realized that the reader does encoding-sniffing.  So if someone saves the file as UTF-16 this would still successfully read it.

If we're willing to say "the file is UTF-8" (or, more likely, "the file is 7-bit ASCII") then this can be simplified a lot.